hols/todo/infra/multi-cluster/README.md

```diff
@@ -0,0 +1,263 @@
+# Multi-region service mesh with Istio and OKE
+
+## Assumptions
+
+1. A pair of OKE clusters in 2 different OCI regions will be used.
+2. The OKE clusters will use private control planes.
+3. The topology model used is [Multi-Primary on different networks](https://istio.io/latest/docs/setup/install/multicluster/multi-primary_multi-network/).
+
+![Multi-primary on multiple networks](docs/assets/multi-primary%20multi-networks.png)
+4. This example uses self-signed certificates.
+
+## Create the OKE Clusters
+
+1. Copy the terraform.tfvars.example to terraform.tfvars and provide the necessary values as detailed in steps 2-6.
+
+2. Configure the provider parameters:
+
+```
+# provider
+api_fingerprint = ""
+
+api_private_key_path = "~/.oci/oci_rsa.pem"
+
+home_region = "ashburn"
+
+tenancy_id = "ocid1.tenancy.oc1.."
+
+user_id = "ocid1.user.oc1.."
+
+compartment_id = "ocid1.compartment.oc1.."
+```
+
+3. Configure an ssh key pair:
+
+```
+# ssh
+ssh_private_key_path = "~/.ssh/id_rsa"
+ssh_public_key_path  = "~/.ssh/id_rsa.pub"
+```
+
+4. Configure your clusters' regions.
+
+```
+# clusters
+clusters = {
+  c1 = { region = "sydney", vcn = "10.1.0.0/16", pods = "10.201.0.0/16", services = "10.101.0.0/16", enabled = true }
+  c2 = { region = "melbourne", vcn = "10.2.0.0/16", pods = "10.202.0.0/16", services = "10.102.0.0/16", enabled = true }
+}
+```
+
+5. Configure additional parameters if necessary:
+
+```
+kubernetes_version = "v1.30.1"
+
+cluster_type = "basic"
+
+oke_control_plane = "private"
+```
+
+6. Configure your node pools:
+
+```
+nodepools = {
+  np1 = {
+    shape            = "VM.Standard.E4.Flex",
+    ocpus            = 2,
+    memory           = 64,
+    size             = 2,
+    boot_volume_size = 150,
+  }
+}
+```
+
+7. Run terraform to create your clusters:
+
+```
+terraform apply --auto-approve
+```
+
+8. Once the Dynamic Routing Gateways (DRGs) and Remote Peering Connections (RPCs) have been created, use the OCI console to establish a connection between them.
+
+## Install Istio
+
+1. Terraform will output an ssh convenience command. Use it to ssh to the operator host:
+
+```
+ssh_to_operator = "ssh -o ProxyCommand='ssh -W %h:%p -i ~/.ssh/id_rsa opc@<bastion_ip>' -i ~/.ssh/id_rsa opc@<operator_ip>"
+```
+
+2. Verify connectivity to both clusters:
+
+```
+for cluster in c1 c2; do
+  ktx $cluster
+  k get nodes
+done
+```
+
+3. Generate certs for each cluster:
+
+```
+export ISTIO_HOME=/home/opc/istio-1.23.0
```

Comment:
Do we document the pre-req that installs `/home/opc/istio` ?

Reply:
It's done by terraform. The user doesn't do anything